Backpack/FlightLeg/Classes/BPKFlightLeg.swift

```diff
@@ -0,0 +1,242 @@
+        stackView.alignment = .trailing
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        return stackView
+    }()
+    
+    private let flightTimeStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.alignment = .leading
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        return stackView
+    }()
+    
+    public init(
+        departureArrivalTime: String,
+        nextDayArrival: String?,
+        flightDescription: NSAttributedString,
+        stopsInfo: String,
+        highlightStopsInfo: Bool,
+        duration: String,
+        operatedBy: String?,
+        warning: String?,
+        carrierLogo: BPKIconName
+    ) {
+        self.departureArrivalTime = departureArrivalTime
+        self.nextDayArrival = nextDayArrival
+        self.flightDescription = flightDescription
+        self.stopsInfo = stopsInfo
+        self.highlightStopsInfo = highlightStopsInfo
+        self.duration = duration
+        self.operatedBy = operatedBy
+        self.warning = warning
+        self.carrierLogo = carrierLogo
+        
+        super.init(frame: .zero)
+        setupView()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setupView() {
+        containerStackView.addArrangedSubview(setupCarrierLogo())
+        containerStackView.addArrangedSubview(setupFlightStackView())
+        containerStackView.addArrangedSubview(setupDurationStackView())
+        
+        addSubview(containerStackView)
+        
+        NSLayoutConstraint.activate([
+            containerStackView.topAnchor.constraint(equalTo: topAnchor),
+            containerStackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            containerStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            containerStackView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+    }
+    
+    private func setupCarrierLogo() -> UIView {
+        let verticalStackView = UIView()
+        verticalStackView.translatesAutoresizingMaskIntoConstraints = false
+        
+        carrierLogoIcon.iconName = carrierLogo
+        leadingStackView.addArrangedSubview(carrierLogoIcon)
+        verticalStackView.addSubview(leadingStackView)
+        
+        NSLayoutConstraint.activate([
+            leadingStackView.topAnchor.constraint(equalTo: verticalStackView.topAnchor, constant: BPKSpacingSm),
+            leadingStackView.bottomAnchor.constraint(equalTo: verticalStackView.bottomAnchor),
+            leadingStackView.leadingAnchor.constraint(equalTo: verticalStackView.leadingAnchor),
+            leadingStackView.trailingAnchor.constraint(equalTo: verticalStackView.trailingAnchor)
+        ])
+        return verticalStackView
+    }
+    
+    private func setupFlightStackView() -> UIView {
```

Comment:
can you rename this function to reflect that's actually creating flight stack? something like `createFlightStackView` or `makeFlig...`